src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -826,8 +826,9 @@ string Cci.INamespaceTypeReference.NamespaceName
                 // INamespaceTypeReference is a type contained in a namespace
                 // if this method is called for a nested type, we are in big trouble.
                 Debug.Assert(((Cci.ITypeReference)this).AsNamespaceTypeReference != null);
-                return this.ContainingSymbol.ToDisplayString(SymbolDisplayFormat.QualifiedNameOnlyFormat);
-            }
+
+                return this.ContainingNamespace.QualifiedName;
```

Comment:
If that is a problem for VB. I did not look at profiles for VB yet. It may or may not use this frequently enough.